bundles/binding/org.openhab.binding.akm868/src/main/java/org/openhab/binding/akm868/internal/AKM868Binding.java

```diff
@@ -0,0 +1,205 @@
+		logger.debug("Stoppig AKM868 listener...");
+		if (packetlistener != null)
+			packetlistener.stopListener();
+		// deallocate resources here that are no longer needed and 
+		// should be reset when activating this binding again
+	}
+
+	
+	/**
+	 * @{inheritDoc}
+	 */
+	@Override
+	protected long getRefreshInterval() {
+		return refreshInterval;
+	}
+
+	/**
+	 * @{inheritDoc}
+	 */
+	@Override
+	protected String getName() {
+		return "AKM868 Refresh Service";
+	}
+	
+	/**
+	 * @{inheritDoc}
+	 */
+	@Override
+	protected void execute() {
+								
+	}
+
+	/**
+	 * @{inheritDoc}
+	 */
+	@Override
+	protected void internalReceiveCommand(String itemName, Command command) {
+		// the code being executed when a command was sent on the openHAB
+		// event bus goes here. This method is only called if one of the 
+		// BindingProviders provide a binding for the given 'itemName'.
+//		logger.debug("AKM 868: internalReceiveCommand() is called!");
+	}
+	
+	/**
+	 * @{inheritDoc}
+	 */
+	@Override
+	protected void internalReceiveUpdate(String itemName, State newState) {
+		// the code being executed when a state was sent on the openHAB
+		// event bus goes here. This method is only called if one of the 
+		// BindingProviders provide a binding for the given 'itemName'.
+//		logger.debug("AKM 868:internalReceiveUpdate() is called!");
+	}
+		
+	/**
+	 * @{inheritDoc}
+	 */
+	@Override
+	public void updated(Dictionary<String, ?> config) throws ConfigurationException {
+		if (config != null) {
+			
+			// to override the default refresh interval one has to add a 
+			// parameter to openhab.cfg like <bindingName>:refresh=<intervalInMs>
+			String refreshIntervalString = (String) config.get("refresh");
+			if (StringUtils.isNotBlank(refreshIntervalString)) {
+				refreshInterval = Long.parseLong(refreshIntervalString);
+			}
+			
+			String hostString = (String) config.get("host");
+			if (StringUtils.isNotBlank(hostString)) {
+				host = hostString;
+			}
+			String portString = (String) config.get("port");
+			if (StringUtils.isNotBlank(portString)) {
+				port = Integer.parseInt(portString);
+			}
+			
+			String timeoutString = (String) config.get("timeout");
+			if (StringUtils.isNotBlank(timeoutString)) {
+				timeout = Long.parseLong(timeoutString); 
+			}
+			
+			// make sure that there is no listener running
+			packetlistener.stopListener();
+			// send the parsed information to the listener
+			packetlistener.initializeReceiver(host,port,timeout);
+			// start the listener
+			new Thread(packetlistener).start();
+			setProperlyConfigured(true);
+		}
+	}
+
+
+	@Override
+	public void publishUpdate(String id, boolean isPresent) {
+		
+		for (AKM868BindingProvider provider : providers) {
+			for (String itemName : provider.getItemNames()) {
+				
+				if (provider.getId(itemName).equals(id) && provider.getChannel(itemName).equals("0")) {
+					logger.debug("Publishing state for: "+itemName+" -state: "+provider.getId(itemName));
```

Comment:
I'll offer: using string concatenation with + signs causes memory and processor time, even if the debug message isn't printed.  If you instead say 
```
logger.debug("Publishing state for: {} -state: {}", itemName, provider.getId(itemName));
```
then you incur no real overhead unless the debug message is printed.